app/grandchallenge/algorithms/serializers.py

```diff
@@ -90,3 +94,85 @@ class HyperlinkedJobSerializer(JobSerializer):
 
     class Meta(JobSerializer.Meta):
         pass
+
+
+class JobPostSerializer(JobSerializer):
+    algorithm_title = CharField(write_only=True)
+    inputs = ComponentInterfaceValuePostSerializer(many=True)
+
+    class Meta:
+        model = Job
+        fields = ["pk", "algorithm_title", "inputs", "status"]
+
+    def validate(self, data):
+        alg = Algorithm.objects.get(title=data.pop("algorithm_title"))
```

Comment:
Thinking about it, it might be better to have this functionality as a POST action on the algorithm itself? That's how we structure it in the Django template views. Unsure.

Reply:
It's a bit strange, as you post a job, and the response is also a job. So for the api I think this makes more sense actually.